nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/controller/reporting/LogComponentStatuses.java

```diff
@@ -0,0 +1,205 @@
+
+    private final String processorHeader;
+    private final String processorBorderLine;
+    private final String counterHeader;
+    private final String counterBorderLine;
+
+    private final Map<String, Long> previousCounterValues = new ConcurrentHashMap<>();
+    private volatile long lastTriggerTime = System.currentTimeMillis();
+
+    public LogComponentStatuses(final FlowFileEventRepository flowFileEventRepository, final CounterRepository counterRepository, final FlowManager flowManager) {
+        this.flowFileEventRepository = flowFileEventRepository;
+        this.counterRepository = counterRepository;
+        this.flowManager = flowManager;
+
+        processorHeader = String.format(PROCESSOR_LINE_FORMAT, "Processor Name", "Processor ID", "Processor Type", "Bytes Read/sec", "Bytes Written/sec", "Tasks/sec", "Nanos/Task",
+            "Percent of Processing Time");
+        processorBorderLine = createLine(processorHeader);
+
+        counterHeader = String.format(COUNTER_LINE_FORMAT, "Counter Context", "Counter Name", "Counter Value", "Increase/sec");
+        counterBorderLine = createLine(counterHeader);
+    }
+
+    private String createLine(final String valueToUnderscore) {
+        final StringBuilder processorBorderBuilder = new StringBuilder(valueToUnderscore.length());
+        for (int i = 0; i < valueToUnderscore.length(); i++) {
+            processorBorderBuilder.append('-');
+        }
+        return processorBorderBuilder.toString();
+    }
+
+    @Override
+    public void run() {
+        try {
+            if (!logger.isInfoEnabled()) {
+                return;
+            }
+
+            logFlowFileEvents();
+            logCounters();
+        } catch (final Exception e) {
+            logger.error("Failed to log component statuses", e);
+        }
+    }
+
+    private void logFlowFileEvents() {
+        final long timestamp = System.currentTimeMillis();
+        final ProcessGroup rootGroup = flowManager.getRootGroup();
+        final List<ProcessorNode> allProcessors = rootGroup.findAllProcessors();
+
+        long totalNanos = 0L;
+        final List<ProcessorAndEvent> processorsAndEvents = new ArrayList<>();
+        for (final ProcessorNode processorNode : allProcessors) {
+            final FlowFileEvent flowFileEvent = flowFileEventRepository.reportTransferEvents(processorNode.getIdentifier(), timestamp);
+            if (flowFileEvent == null) {
+                continue;
+            }
+
+            processorsAndEvents.add(new ProcessorAndEvent(processorNode, flowFileEvent));
+            totalNanos += flowFileEvent.getProcessingNanoseconds();
+        }
+
+        final Comparator<ProcessorAndEvent> comparator = Comparator.comparing(procAndEvent -> procAndEvent.getEvent().getProcessingNanoseconds());
+        processorsAndEvents.sort(comparator.reversed());
+
+        final StringBuilder builder = new StringBuilder();
+        builder.append("Processor Statuses:\n");
```

Comment:
Although using the newline character seems acceptable is most cases, would it make sense to replace the occurrences of `\n` with `System.lineSeparator()`?

Reply:
Good point about existing usage, in that case, leaving it as-is sounds good.